File: composeApp/src/androidMain/kotlin/com/dnfapps/arrmatey/entensions/EnumStringExtensions.kt
package com.dnfapps.arrmatey.entensions

import androidx.annotation.StringRes
import com.dnfapps.arrmatey.R
import com.dnfapps.arrmatey.api.arr.model.MovieStatus
import com.dnfapps.arrmatey.api.arr.model.SeriesMonitorType
import com.dnfapps.arrmatey.api.arr.model.SeriesType
import org.jetbrains.compose.resources.StringResource

@StringRes
fun MovieStatus.stringResource() = when (this) {
    MovieStatus.Tba -> R.string.tba
    MovieStatus.Deleted -> R.string.deleted
    MovieStatus.Released -> R.string.released
    MovieStatus.Announced -> R.string.announced
    MovieStatus.InCinemas -> R.string.in_cinemas
}

fun SeriesMonitorType.stringResource() = when (this) {
    SeriesMonitorType.Unknown -> R.string.unknown
    SeriesMonitorType.All -> R.string.all
    SeriesMonitorType.Future -> R.string.future
    SeriesMonitorType.Missing -> R.string.missing
    SeriesMonitorType.Existing -> R.string.existing
    SeriesMonitorType.FirstSeason -> R.string.first_season
    SeriesMonitorType.LastSeason -> R.string.last_season
    SeriesMonitorType.LatestSeason -> R.string.latest_seasons
    SeriesMonitorType.Pilot -> R.string.pilot
    SeriesMonitorType.Recent -> R.string.recent
    SeriesMonitorType.UnmonitorSpecials -> R.string.unmonitor_specials
    SeriesMonitorType.MonitorSpecials -> R.string.monitor_specials
    SeriesMonitorType.None -> R.string.none
    SeriesMonitorType.Skip -> R.string.skip
}

fun SeriesType.stringResource() = when (this) {
    SeriesType.Standard -> R.string.standard
    SeriesType.Daily -> R.string.daily
    SeriesType.Anime -> R.string.anime
}

File: iosApp/iosApp/Extensions/EnumStringExtensions.swift
//
//  EnumStringExtensions.swift
//  iosApp
//
//  Created by Owen LeJeune on 2025-12-31.
//

import Shared

extension MovieStatus {
    func label() -> String {
        switch self {
        case .tba: "tba"
        case .deleted: "deleted"
        case .released: "released"
        case .announced: "announced"
        case .inCinemas: "in_cinemas"
        }
    }
}

extension SeriesMonitorType {
    func label() -> String {
        switch self {
        case .unknown: "unknown"
        case .all: "all"
        case .future: "future"
        case .missing: "missing"
        case .existing: "existing"
        case .firstSeason: "first_season"
        case .lastSeason: "last_season"
        case .latestSeason: "latest_season"
        case .pilot: "pilot"
        case .recent: "recent"
        case .monitorSpecials: "monitor_specials"
        case .unmonitorSpecials: "unmonitor_specials"
        case .none: "none"
        case .skip: "skip"
        }
    }
}

extension SeriesType {
    func label() -> String {
        switch self {
        case .standard: "standard"
        case .daily: "daily"
        case .anime: "anime"
        }
    }
}
